tensorflow/python/ops/variable_scope.py

```diff
@@ -652,6 +652,10 @@ def _get_single_variable(self,
       raise ValueError("Shape of a new variable (%s) must be fully defined, "
                        "but instead was %s." % (name, shape))
 
+    # Fill the integer array with zeros
+    if initializer is None and dtype.is_integer:
+      initializer = init_ops.zeros_initializer(shape=shape, dtype=dtype.base_dtype)
```

Comment:
Wouldn't it be better for this to be a function instead? In that case, for partitioned variables and such, there would be better support.